### PR TITLE
MGDAPI-1141

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## JIRA 
https://issues.redhat.com/browse/MGDAPI-1141

## Changes
- update image mirroring registry
- update prow config generation to be release specific
- fix some typos

### CODE REVIEW

**Review by Boomatang — COMMENTED** [review]
It all looks good but I do have two questions.
1) Why was the context removed from some of the function signatures? 
2) How can I verify the the command output is correct?

File: cmd/openshiftCIAddBranch.go [hunk 1]
```diff
@@ -148,11 +165,14 @@ func init() {
 	cmd.MarkFlagRequired("branch")
 	cmd.Flags().StringVar(&f.repoDir, "repo-dir", "", "Repo Dir")
 	cmd.MarkFlagRequired("repo-dir")
+	cmd.Flags().StringVar(&f.repoDir, "olm-type", "", "Repo Dir")
+	cmd.MarkFlagRequired("olm-type")
```
Comment (Boomatang) [inline cmd/openshiftCIAddBranch.go:169]: We are ignoring the error returned from the `MarkFlagRequired`. Should we add helpful error messages for these flags.

Comment (Boomatang) [reply cmd/openshiftCIAddBranch.go:169]: Looking at this a second time the `olm-type` needs a better description

**Comment by briangallagher** [comment]
@laurafitzgerald I haven't ran the command but the code changes look good to me.

**Comment by laurafitzgerald** [comment]
> It all looks good but I do have two questions.
> 
> 1. Why was the context removed from some of the function signatures?
> 2. How can I verify the the command output is correct?

Hi @Boomatang 

1) I removed context as it was unused in the function it was passed to. I removed other passed variables from function signatures because they are available in the struct. 

2) Thanks for the review. I will add verification steps to the pr now. 👍

File: cmd/openshiftCIAddBranch.go (hunk 1 above)
Comment (laurafitzgerald) [reply cmd/openshiftCIAddBranch.go]: Yes, well spotted. I've improved the definition of the param. It also needed to be updated to match the param as passed here -> https://gitlab.cee.redhat.com/integreatly-qe/ci-cd/-/blob/master/pipelines/rhmi-release.groovy#L134

Comment (laurafitzgerald) [reply cmd/openshiftCIAddBranch.go]: Does it matter that it's also defined by the parent command? https://github.com/integr8ly/delorean/blob/master/cmd/root.go#L115

Comment (laurafitzgerald) [reply cmd/openshiftCIAddBranch.go]: I'm also just noticing now that the command being used in the pipeline is `
`delorean release openshift-ci-release` whereas I've updated `delorean release add-branch` 
To me it looks like the code is completely duplicated and `release add-branch` is not referenced in the release pipeline. This is strange. I will need to push further changes.

**Comment by laurafitzgerald** [comment]
@Boomatang this one is ready for a review again. Most of the new changes are tests but if you want to run through it review wise first we can pair on that and then I will add verification steps.

**Comment by Boomatang** [comment]
From our call I would like to make a note of the few things we talk about. Please correct me if I missed any thing. 
1. We question the need for the `openshift-ci add-branch` when it seem to be the very same as the `release openshift-ci-release` command. Looking into pulling the common logic out of the commands. But not remove any of the cli commands.
2. It was clarified these command should only ever be ran on release branches. This question raised with looking at the defaults for the switch statements.
3.  The test case `TestOSDAddonRelease` is been a bit more readable 
4. There was talk about deprecating the `NewRHMIVersion` function. It may be some time before it can be fully removed.
5. Talks about bring urls strings out to some kind of config or types file

**Comment by laurafitzgerald** [comment]
Hi @Boomatang thanks for summarising the discussion we had Friday.
On 1. I investigated consolidating the logic but there are some small differences so seeing as it's only two locations I've updated the file paths and the internal org so that they are declared only in one location which addresses what you've said in 5. Would you mind taking a look at the new change. I've pushed it in a separate commit.

**Comment by laurafitzgerald** [comment]
I've run a test for this using this command -> 
./delorean release openshift-ci-release \
>               --owner integr8ly \
>               --repo integreatly-operator \
>               --ci-org-origin integr8ly \
>               --ci-org-upstream openshift \
>               --ci-repo release \
>               --branch master \
>               --version 99.99.99 \
>               --olmType managed-api-service \
>               --user <redacted> \
>               --token <redacted>```

File: cmd/openshiftCIAddBranch.go [hunk 2]
```diff
@@ -12,6 +12,11 @@ import (
 	"strings"
 )
 
+const ProwConfigSourceRHMI = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-release-v2.0.yaml"
+const ProwConfigSourceRHOAM = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-rhoam-release-v1.1.yaml"
+const ProwConfigSourceMaster = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-master.yaml"
+const ProwInternalRegistry = "registry.ci.openshift.org/integr8ly"
```
Comment (Boomatang) [inline cmd/openshiftCIAddBranch.go:18]: Possible the format here should be 
```
const (
	ProwConfigSourceRHMI   = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-release-v2.0.yaml"
	ProwConfigSourceRHOAM  = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-rhoam-release-v1.1.yaml"
	ProwConfigSourceMaster = "ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-master.yaml"
	ProwInternalRegistry   = "registry.ci.openshift.org/integr8ly"
)
```

**Review by Boomatang — COMMENTED** [review]
Apart from the one failing unit test i am happy, The sample PR looks correct.

**Comment by laurafitzgerald** [comment]
Also tested that RHMI would work as expected

Running this command 
```
./delorean release openshift-ci-release \
              --owner integr8ly \
              --repo integreatly-operator \
              --ci-org-origin integr8ly \
              --ci-org-upstream openshift \
              --ci-repo release \
              --branch master \
              --version 99.99.99 \
              --olmType integreatly-operator \
              --user <redacted> \
              --token <redacted> 
```

produced this pr -> https://github.com/openshift/release/pull/15443

**Comment by laurafitzgerald** [comment]
Verification of change to `openshift-ci add-branch` 

Ran 
`../delorean openshift-ci add-branch --branch 3scale-next-9.9.9 --repo-dir ~/openshift/release `

image mirroring as expected
```
cat core-services/image-mirroring/integr8ly/mapping_integr8ly_operator_3scale-next-9_9_9
registry.ci.openshift.org/integr8ly/3scale-next-9.9.9:integreatly-operator quay.io/integreatly/integreatly-operator:3scale-next-9.9.9
registry.ci.openshift.org/integr8ly/3scale-next-9.9.9:integreatly-operator-test-harness quay.io/integreatly/integreatly-operator-test-harness:3scale-next-9.9.9

```

prow config files as expected
```
 ci-operator/config/integr8ly/integreatly-operator/integr8ly-integreatly-operator-3scale-next-9.9.9.yaml
        ci-operator/jobs/integr8ly/integreatly-operator/integr8ly-integreatly-operator-3scale-next-9.9.9-postsubmits.yaml
        ci-operator/jobs/integr8ly/integreatly-operator/integr8ly-integreatly-operator-3scale-next-9.9.9-presubmits.yaml

```

**Comment by laurafitzgerald** [comment]
@Boomatang I removed the inclusion of the olmtype from the openshift-ci add-branch and just updated it to use the centralised strings now. I don't think it's necessary to have it there as it's only used ews cases where the source would be master anyway. I included the verification steps I took above. 
If you could throw a final eye on this that would be great.

**Comment by Boomatang** [comment]
Yap this change makes sense when you see the resulting output. Are we sure this is only used the ews?
/lgtm

**Comment by laurafitzgerald** [comment]
@Boomatang I actually can't find any reference to it. I pinged Pavel and Mike about it in the chat but I think the way it is now won't break anything but still centralises the file paths and registry link. If you are good with this I will post the pr for approval so it can merge now.

**Comment by Boomatang** [comment]
Yap I am still good with it. That function is not used in the release pipeline so we are not breaking that. I think we can approve the pr.

**Pull request merged**